consensus/src/block_storage/block_tree.rs

```diff
@@ -298,6 +305,7 @@ where
     /// Note that we do not necessarily remove the pruned blocks: they're kept in a separate buffer
     /// for some time in order to enable other peers to retrieve the blocks even after they've
     /// been committed.
+    // REVIEW: How long do we need to persist them? Is this for correctness or an optimization?
```

Comment:
it's optimization to help nodes fall behind so that it won't stay behind forever like need to fetch block -> already committed -> sync -> need to fetch block ->

Reply:
It might be worth stress testing this by having a config where we are strict about deletion.

This seems like a way we could inadvertently cause a liveness problem (if we end up growing a liveness dependency on having this data / a corner case make us aggressive about deletion)

Reply:
agree it's worth testing, it's not a liveness problem but rather fairness problem. 
we can't have more than 1/3 nodes fall behind, at that time they'd have chance to sync and get all the blocks.
but when there's only handful of nodes fall behind, they may stay behind until majority slows down